### PR TITLE
Added signature templating code.

### PR DESCRIPTION
**Pull request by xvzcf**
* Added the ability to automatically add new libOQS signature schemes to OpenSSL by specifying various algorithm data in `oqs-template/generate.yml`
* Used this to add support for the `oqsdefault` signature method and rearrange and rename the signature schemes.

### CODE REVIEW

**Comment by christianpaquin** [comment]
Great, I'll take a look tomorrow.

**Comment by christianpaquin** [comment]
PR looks good. I scanned the diff and built/tested the lib on linux. One issue I faced in the past was some out-of-order alg definitions; I've marked the troublesome areas with "keep in NID order" comments in `crypto/asn1/standard_methods.h` and  `crypto/evp/pmeth_lib.c`. I see that these comments have been moved to the corresponding script fragments. I'm not exactly sure how the templating scripts work, so I assume you handled them well (where is this documented? e.g., how to add a new alg).

**Comment by xvzcf** [comment]
Each fragment iterates through the entries specified under `sigs` in `generate.yml`. `sigs` is an ordered collection, so the fragments will print entries in the same order in which they were specified in the `generate.yml`. 

The comments are therefore not strictly necessary, but I moved them over to the fragments as they do constitute useful knowledge about the codebase and might come in handy if the need for some special processing arises in the future.

I will also update the README (in a separate PR) to specify this new method of adding key-exchanges and signatures.

**Comment by christianpaquin** [comment]
Great. And the circleci tests run all the integration previously on the testing project, correct @dstebila?

**Comment by dstebila** [comment]
> Great. And the circleci tests run all the integration previously on the testing project, correct @dstebila?

Yes, that's correct.

File: ssl/t1_lib.c [hunk 1]
```diff
@@ -2629,23 +2649,27 @@ void tls1_set_cert_validity(SSL *s)
     tls1_check_chain(s, NULL, NULL, NULL, SSL_PKEY_ED25519);
     tls1_check_chain(s, NULL, NULL, NULL, SSL_PKEY_ED448);
 #if !defined(OQS_NIST_BRANCH)
```
Comment (dstebila) [inline ssl/t1_lib.c:2651]: Can delete the NIST\_BRANCH check now?

**Review by dstebila — APPROVED** [review]

File: ssl/t1_lib.c (hunk 1 above)
Comment (xvzcf) [reply ssl/t1_lib.c:2651]: Working on it.

**Pull request merged**